contracts/test/umbra-hook-test.js

```diff
@@ -0,0 +1,407 @@
+    expect(callData.acceptor).to.equal(acceptor);
+    expect(callData.tokenAddr).to.equal(ctx.token.address);
+    expect(callData.sponsor).to.equal(AddressZero);
+    expect(fromWei(callData.sponsorFee)).to.equal('0');
+    expect(callData.data).to.equal('0xbeefc0ffee');
+
+    const balance = await ctx.token.balanceOf(acceptor);
+    expect(fromWei(balance)).to.equal('100');
+  });
+
+  it('should call the hook contract when the stealth receiver withdraws directly with empty data', async () => {
+    await mintAndSendToken(sender, receiver, '100');
+
+    await ctx.umbra.withdrawTokenAndCall(acceptor, ctx.token.address, ctx.hook.address, [], {
+      from: receiver,
+    });
+
+    const callData = await ctx.hook.lastCallData();
+
+    expect(fromWei(callData.amount)).to.equal('100');
+    expect(callData.stealthAddr).to.equal(receiver);
+    expect(callData.acceptor).to.equal(acceptor);
+    expect(callData.tokenAddr).to.equal(ctx.token.address);
+    expect(callData.sponsor).to.equal(AddressZero);
+    expect(fromWei(callData.sponsorFee)).to.equal('0');
+    expect(callData.data).to.equal(null);
+
+    const balance = await ctx.token.balanceOf(acceptor);
+    expect(fromWei(balance)).to.equal('100');
+  });
+
+  it('should call the hook contract when the stealth receiver withdraws directly to the hook contract', async () => {
+    await mintAndSendToken(sender, receiver, '100');
+
+    await ctx.umbra.withdrawTokenAndCall(ctx.hook.address, ctx.token.address, ctx.hook.address, '0xbeefc0ffee', {
+      from: receiver,
+    });
+
+    const callData = await ctx.hook.lastCallData();
+
+    expect(fromWei(callData.amount)).to.equal('100');
+    expect(callData.stealthAddr).to.equal(receiver);
+    expect(callData.acceptor).to.equal(ctx.hook.address);
+    expect(callData.tokenAddr).to.equal(ctx.token.address);
+    expect(callData.sponsor).to.equal(AddressZero);
+    expect(fromWei(callData.sponsorFee)).to.equal('0');
+    expect(callData.data).to.equal('0xbeefc0ffee');
+
+    const balance = await ctx.token.balanceOf(ctx.hook.address);
+    expect(fromWei(balance)).to.equal('100');
+  });
+
+  it('should not call the hook when the stealth receiver withdraws directly if the hook address is 0', async () => {
+    await mintAndSendToken(sender, receiver, '100');
+
+    await ctx.umbra.withdrawTokenAndCall(acceptor, ctx.token.address, AddressZero, '0xbeefc0ffee', {
+      from: receiver,
+    });
+
+    const callData = await ctx.hook.lastCallData();
+
+    // call data in the mock trigger should still be zeroes
+    expect(fromWei(callData.amount)).to.equal('0');
+    expect(callData.stealthAddr).to.equal(AddressZero);
+    expect(callData.acceptor).to.equal(AddressZero);
+    expect(callData.tokenAddr).to.equal(AddressZero);
+    expect(callData.sponsor).to.equal(AddressZero);
+    expect(fromWei(callData.sponsorFee)).to.equal('0');
+    expect(callData.data).to.equal(null);
+
+    // the withdrawal  transfer should still work
```

Comment:
Found an extra space 😈

Which reminds me—before merging, just run `yarn prettier` from the repo root just in case that wasn't fixed properly

Reply:
I don't know how you catch these!

I'd like to hold off on running prettier to avoid a bunch of unrelated diff noise on this PR. Will open a separate PR for that after this one is merged.

I also don't understand why it's apparently not running on the commit hooks, despite it appearing to do something 🤷‍♂️